pandas/core/indexing.py

```diff
@@ -1881,14 +1850,34 @@ def _getitem_axis(self, key, axis: int):
         self._validate_key(key, axis)
         return self._get_label(key, axis=axis)
 
+    def _get_slice_axis(self, slice_obj: slice, axis: int):
+        """
+        This is pretty simple as we just have to deal with labels.
+        """
+        # caller is responsible for ensuring non-None axis
+        obj = self.obj
+        if not need_slice(slice_obj):
+            return obj.copy(deep=False)
+
+        labels = obj._get_axis(axis)
+        indexer = labels.slice_indexer(
+            slice_obj.start, slice_obj.stop, slice_obj.step, kind=self.name
+        )
+
+        if isinstance(indexer, slice):
+            return self._slice(indexer, axis=axis, kind="iloc")
+        else:
```

Comment:
hmm this condition should likely be handled in ``_slice`` itself

Reply:
yah, _slice is a one-liner that just calls self.obj._slice, so i plan to get rid of it.  for now this is just putting the methods in more reasonable place unchanged

Reply:
kk